lib/puppet/functions/compare.rb

```diff
@@ -0,0 +1,125 @@
+# Compares two values and returns -1, 0 or 1 if first value is smaller, equal or larger than the second value.
+# The compare function accepts arguments of the data types `String`, `Numeric`, `Timespan`, `Timestamp`, and `Version`,
+# such that:
+#
+# * two of the same data type can be compared
+# * `Timespan` and `Timestamp` can be compared with each other and with `Numeric`
+#
+# When comparing two `String` values the comparison can be made to consider case by passing a third (optional)
+# boolean `false` value - the default is `true` which ignores case as the comparison operators
+# in the Puppet Language.
+#
+Puppet::Functions.create_function(:compare) do
```

Comment:
You need this function since the compare operators return boolean and not -1, 0, 1 which the compare function does. You use it to compare for things like max, min, sort that takes a lambda and you want to do a comparison.